vmx/src/vm.rs

```diff
@@ -213,6 +146,80 @@ impl VM {
     pub fn session(&mut self) -> Session {
         Session::new(self.clone())
     }
+
+    pub(crate) fn memory_path(
+        &self,
+        module_id: &ModuleId,
+    ) -> (MemoryPath, MemoryFreshness) {
+        (
+            MemoryPath::new(
+                self.inner
+                    .read()
+                    .base_memory_path
+                    .join(module_id_to_name(*module_id)),
+            ),
+            Fresh,
+        )
+    }
+
+    pub(crate) fn path_to_module_commit(
+        &self,
+        module_id: &ModuleId,
+        module_commit_id: &ModuleCommitId,
+    ) -> MemoryPath {
+        const SEPARATOR: char = '_';
+        let commit_id_name = &*commit_id_to_name(*module_commit_id);
+        let mut name = module_id_to_name(*module_id);
+        name.push(SEPARATOR);
+        name.push_str(commit_id_name);
+        let path = self.inner.read().base_memory_path.join(name);
+        MemoryPath::new(path)
+    }
+
+    pub(crate) fn path_to_session_commit(
+        &self,
+        module_id: &ModuleId,
+        session_commit_id: &SessionCommitId,
+    ) -> Option<MemoryPath> {
+        self.get_module_commit_id(session_commit_id, module_id).map(
+            |module_commit_id| {
+                self.path_to_module_commit(module_id, &module_commit_id)
+            },
+        )
+    }
+
+    pub(crate) fn add_session_commit(&mut self, session_commit: SessionCommit) {
+        self.inner.write().session_commits.add(session_commit);
+    }
+
+    pub(crate) fn restore_session(
```

Comment:
This is a bit of a conceptual issue, i think it would be more reasonable to have a VM::from_commit_id(...); - the `SessionCommitId` is redundant and should be called `CommitId`

Reply:
I cannot create VM out of session id as it is VM who has the location of all session's persistence, the base location..

Reply:
ok let's keep it like this for now